.editorconfig

```diff
@@ -0,0 +1,133 @@
+# editorconfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Default settings:
+# A newline ending every file
+# Use 4 spaces as indentation
+[*]
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+
+# C# files
+[*.cs]
+# New line preferences
+csharp_new_line_before_open_brace = all
+csharp_new_line_before_else = true
+csharp_new_line_before_catch = true
+csharp_new_line_before_finally = true
+csharp_new_line_before_members_in_object_initializers = true
+csharp_new_line_before_members_in_anonymous_types = true
+csharp_new_line_between_query_expression_clauses = true
+
+# Indentation preferences
+csharp_indent_block_contents = true
+csharp_indent_braces = false
+csharp_indent_case_contents = true
+csharp_indent_switch_labels = true
+csharp_indent_labels = one_less_than_current
+
+# avoid this. unless absolutely necessary
+dotnet_style_qualification_for_field = false:suggestion
+dotnet_style_qualification_for_property = false:suggestion
+dotnet_style_qualification_for_method = false:suggestion
+dotnet_style_qualification_for_event = false:suggestion
+
+# only use var when it's obvious what the variable type is
```

Comment:
Type inference is a good thing; I'm in the "var all the things" camp.  There's plenty of semantic information around typing in modern IDEs.

Reply:
Type inference may be a good thing when you are writing the code and perhaps acceptable if you are working with the code in a modern IDE that allows you to easy jump around the code and see the actual types. However personally I do not believe type inference is a good thing for the long-term maintenance and readability of the code when you are viewing the code on github or just looking at it as normal text. From my experiences I would never recommend using var unless it is clear what the type of the variable is by simply reading the text of the code. There is a lot of information to be had by knowing the type of the variables to enable you to dig deeper if necessary but not knowing the type of a variable you end up having to dig backwards as opposed to moving forward when reading through the code, it adds more work and takes longer to understand the code. Having the types helps with self-documenting the code which helps with understanding the code.

Reply:
I agree with this statement at a conceptual level: code that is clear, expressive, and understandable when being read is important and a key goal.

That said, I respectfully disagree that the underlying types are deserving of being a focal point.  Respectfully, if the intent and context of the code isn't clear from member names, code structure, and associated comments, I'd argue that the author has already failed in the fundamentals; the types used aren't going to bridge that gap and rescue things.   I'd double down on that stance and say that it applies even more so to non-editor scenarios, like reading on GitHub.

In my opinion, type awareness should be, first and foremost, a tooling and compiler concern, used in service of compile-time safety and providing of a rich developer experience.  While I'll certainly agree that it is helpful when exploring code to be able inspect and explore types, I believe that needing to have a deep awareness of the types in a given block of code to reason about it is often a sign of poor design, side-effects, and/or a misuse of OO principles.   

In rebuttal, I'd argue that a reader of the code shouldn't need to be jumping around to understand a given block of code.  It should be clear and digestible in its own context.  Though, I would agree that there should be an easily discoverable and logical path to uncover types within that context to peel back the layers of the onion and dig deeper.   

In summary, I agree that the types should be clear, though I believe that should happen naturally as a output of well written and well structured code.  I'm of the opinion that if you need to explicitly call out types that there are most likely underlying problems with the code or design.